examples/prometheus-operator/example.cue

```diff
@@ -0,0 +1,98 @@
+package prometheusOperator
+
+import "github.com/pollypkg/polly/schema:pollyschema"
+
+// NOTE we could make it an expectation that the emit value is
+// a pollyschema.PollyPackage
+prometheusOperator: pollyschema.PollyPackage & {
+	header: {
+		name: "prometheus-operator"
+		uri:  "github.com/pollypkg/polly/examples/prometheus-operator"
+		params: {}
+	}
+	signals: [
+		// List Errors
+		{
+			name: "ListErrors"
+			lang: "promql"
+			params: {job: string}
+			query: """
+                sum by (controller,namespace) (rate(prometheus_operator_list_operations_failed_total{job="\(params.job)"}[10m]))
+                /
+                sum by (controller,namespace) (rate(prometheus_operator_list_operations_total{job="\(params.job)"}[10m]))
+                > 0.4
```

Comment:
Since this is just a signal and not directly an alert, should we have thresholds in it?

Reply:
You're right, it shouldn't. Same for blow.

Reply:
Just for my understanding, where does the value `params: {job: string}` come from? Are the params set externally?

Reply:
> Just for my understanding, where does the value `params: {job: string}` come from? Are the params set externally?

Well, you can ignore this, as it is still an open issue, as I have now learned.. 😁

Reply:
Yeah i have [future handwaving](https://docs.google.com/document/d/1NbppdxoMrNO5Di865t7UUcOkY5eQMqeSjgvyGW8JdZY/edit#heading=h.grastdqpfmhu) that it would be good to include threshold concepts on signals - generalize to a notion of classifiers that would be useful considerably beyond alerting - but that's a ways away now :)

Note that Twitter's [expression concept](https://twitter.github.io/twitter-server/Admin.html#admin-metric-expressions-json) - conceptually analogous to signals - carries threshold declarations.

Reply:
Yeah, I was focused on the present and not the future :smile:

I think it makes sense to have a threshold in the signal sometime in the future, but for that to be useful we would need to abstract/parametrize expression itself. As it is right now, I cannot see value in having a threshold embedded in the signal expression as this heavily restricts signal usability.

Reply:
> we would need to abstract/parametrize expression itself.

Absolutely, it makes the case for/use cases of parameters that much clearer. And it lines up with the distinction i was making on the call yesterday - we can imagine a straightforward case, where there's a field that's just, say, an `int`, and it's expected that the consumer sets the value directly, themselves, without really any intermediate named parameter.

But there's also a potentially more sophisticated case, where we do have an explicit parameter, which is presumably then:
* Run through some computation to decide the final threshold value
* Also quite likely used somewhere else

The example that always come to mind for this is quorum counts in a distributed database. In that case, there's a set of clear categorizations we can make (OK/all nodes healthy, Degraded/some dead but quorum met, Failed/quorum not met), and the only bit of information we need to concretely define those is the size of the cluster.

Reply:
...But yes, totally out of scope for now :)